h264_encoder_core/src/h264_encoder.cpp

```diff
@@ -78,13 +79,39 @@ class H264EncoderImpl
     dst_width_(-1),
     dst_height_(-1),
     convert_ctx_(nullptr),
+    bitrate_(-1),
     fps_num_(-1),
     fps_den_(-1),
     param_(nullptr),
     pic_in_(nullptr)
   {
   }
 
+  AwsError set_param(AVCodec * codec) {
+    param_ = avcodec_alloc_context3(codec);
```

Comment:
Since this function is re-entrant, you need to close and free `param_` if it is not `nullptr` before assigning it a new context.

Reply:
Or instead close, free, and set `param_` to `nullptr` in your `open_codec()` function on failure, and make sure `param_` always comes in here as `nullptr`.

Reply:
Good point, `param_` should be `nullptr` whenever it enters this function, but I added the check just in case.

Reply:
You'll actually need to set `param_` to `nullptr` after line 119 now, or else you'll be doing double free.

I'll suggest setting `param_ = nullptr` on line 119, and just assert that `param_ == nullptr` on line 77, and test to see if it ever triggers.